### DISCUSSION TITLE
Duda uso diff MacOs

### DISCUSSION THREAD

**Discussion by rominaarancibia** [IIC2133-PUC/2024-1#53] [Tarea 0] [answered]
Hola! Me ocurre que al ocupar el comando diff output.txt tests/parte/correct-output.txt por ejemplo, me marca todo el output diferente a pesar que es idéntico al revisarlo manualmente. Leí en el grupo de telegram que diff -b podía ser de ayuda pero mi computador (que es mac) no lo reconoce. Quedo muy atenta. No estaba segura si se podían adjuntar imágenes por acá. Muchas gracias de antemano.

**Comment by sachondo7** [comment]
que raro, yo tengo mac también y si funciona ese comando. Te recomiendo revisar bien que no tengas espacios de mas (porque esos igual te los cuenta como diferentes) y sino intentar comparar con una herramienta externa como puede ser [esta](https://www.diffchecker.com)

**Comment by PabloNavarroY** [comment]
Puedes revisar el manual de tu versión de `diff` ingresando `man diff` en tu terminal. Otra forma de escribir la flag `-b` es `--ignore-space-change`, por si aparece en el manual.

Puedes adjuntar imágenes por acá del resultado de `diff` (pero no de tu código fuente).

Por si acaso, recuerda que estamos ocupando espacios para indentar el output, no tabs.

**Comment by rominaarancibia** [comment]
Hola!, usé man diff y efectivamente es la versión indicada y --ignore-space-change pero lamentablemente me sigue marcando todo diferente. Adjunto fotos de lo que me da al ocupar diff (que me marca todo diferente) versus la imagen al probar mi output.txt generado usando en la herramienta externa donde dice que están iguales. También, ocupo los espacios indicados para indentar (no tabs). Lamentablemente, me resulta imposible ocupar la herramienta externa con los tests hard ya que son muy grandes y me preocupa que no me funcione el diff por la corrección que es automatizada. Saludos 
<img width="1035" alt="Captura de pantalla 2024-03-26 a la(s) 11 58 12 a  m" src="https://github.com/IIC2133-PUC/2024-1/assets/127449220/28b51827-a278-4e7a-a7bd-77dc15284adb">
<img width="1512" alt="Captura de pantalla 2024-03-26 a la(s) 12 00 22 p  m" src="https://github.com/IIC2133-PUC/2024-1/assets/127449220/6ae50377-5808-4b94-8745-92cce64630c2">
<img width="1021" alt="Captura de pantalla 2024-03-26 a la(s) 12 09 53 p  m" src="https://github.com/IIC2133-PUC/2024-1/assets/127449220/f96ebdd4-52c3-404b-acf9-a98856003080">
<img width="1122" alt="Captura de pantalla 2024-03-26 a la(s) 12 22 54 p  m" src="https://github.com/IIC2133-PUC/2024-1/assets/127449220/df8e4be0-4f86-411d-bab0-5b3f19940172">

**Reply by PabloNavarroY** [reply]
¿Puedes subir lo que llevas a tu repositorio?
Quiero ver como estás escribiendo el output.

**Comment by rominaarancibia** [comment]
Lo acabo de subir. Aprovecho de preguntar si es necesario subir los archivos mencionados en la foto a continuación.
<img width="1028" alt="Captura de pantalla 2024-03-26 a la(s) 12 49 51 p  m" src="https://github.com/IIC2133-PUC/2024-1/assets/127449220/913a1943-bb27-4cb4-a139-42a8552de21c">
Gracias

**Reply by PabloNavarroY** [reply]
No, esa es una advertencia que se da cuando intentas agregar algo que está marcado por el .gitignore

**Comment by PabloNavarroY** [comment, marked as answer]
Encontré el error, estás ocupando `printf` en vez de `fprintf. El primero escribe en la terminal y el segundo escribe en un archivo.

https://www.tutorialspoint.com/c_standard_library/c_function_fprintf.htm

**Reply by rominaarancibia** [reply]
Muchas gracias
